ClothesApp/Application/Services/BrandService.cs

```diff
@@ -0,0 +1,101 @@
+using Application.Dtos;
+using Application.Exceptions;
+using Application.Repositories;
+using AutoMapper;
+using Domain.Entities;
+
+namespace Application.Services;
+
+public class BrandService : IBrandService
+{
+    private readonly IBrandsRepository _brandsRepository;
+
+    private readonly IProductsRepository _productsRepository;
+
+    private readonly IMapper _mapper;
+
+    public BrandService(IBrandsRepository brandsRepository, IProductsRepository productsRepository, IMapper mapper)
+    {
+        _brandsRepository = brandsRepository;
+        _productsRepository = productsRepository;
+        _mapper = mapper;
+    }
+
+    public async Task<BrandDto> AddBrand(CreateBrandDto createBrandDto)
+    {
+        var brandExists = await _brandsRepository.DoesBrandExist(createBrandDto.Name);
+
+        if (brandExists)
+        {
+            throw new DatabaseConflictException(Messages.AlreadyExistsConflict);
+        }
+
+        var brand = _mapper.Map<Brand>(createBrandDto);
+        await _brandsRepository.Insert(brand);
+        var brandDto = _mapper.Map<BrandDto>(brand);
+
+        return brandDto;
+    }
+
+    public async Task<BrandDto> UpdateBrand(BrandDto brandDto)
+    {
+        var brand = _mapper.Map<Brand>(brandDto);
```

Comment:
does this even work? entity will be untracked and it's still updated properly?

Reply:
Yes, it's works

Reply:
Ye, missed Update call